### DISCUSSION TITLE
Defining FluidPorts in a PEM Electrolyser model

### DISCUSSION THREAD

**Discussion by MSudmeier** [OpenModelica/OpenModelica#13041] [Q&A] [answered]
Dear OpenModelica experts,

I am quite new to OpenModelica and i am trying to create an OpenModelica model to validate the following study of a PEM electrolyzer [1-s2.0-S2590174524000886-main.pdf](https://github.com/user-attachments/files/17443394/1-s2.0-S2590174524000886-main.pdf)

I am developing the Anode part of te electrolyser, but I get the error below in the declaration part of the FluidPorts. 

![m_flow error](https://github.com/user-attachments/assets/43ef8eea-4b7a-4587-aacb-589981bcc324)

Below you can see the declaration of the FluidPorts. The equations part is not correct yet as i can't connect the FluidPorts to the internal variables. Can anybody help me with this? I am struggling for days now.   

![FluidPort error](https://github.com/user-attachments/assets/1dc831e7-f8bc-452e-960d-6171ce5f5134)

**Comment by casella** [comment]
Do you get this error when you check the model? Maybe that's just a bug of the checkModel function. Can you share the code so we can reproduce the problem?

**Comment by perost** [comment, marked as answer]
`Modelica.Media.Water` is wrong, it should be e.g. `Modelica.Media.Water.StandardWater`.

**Reply by casella** [reply]
Of course, I did not spot it 😅
